.github/scripts/check_content.py

```diff
@@ -0,0 +1,57 @@
+import os.path
+from os import walk
+
+
+def check_content() -> int:
+    """
+    Walks through the filetree rooted at the current working directory.
+    For each file that it finds, it validates the file
+    """
+    error_count = 0
+    DIRECTORIES_TO_EXCLUDE = ['.git', '.github', '.idea', 'venv', 'DO NOT COMMIT']
+    FILES_TO_EXCLUDE = ['.DS_Store', '.gitignore']
+
+    for root, dirs, files in walk('.', topdown=True):
+        dirs[:] = [d for d in dirs if d not in DIRECTORIES_TO_EXCLUDE]
+        files[:] = [f for f in files if f not in FILES_TO_EXCLUDE]
+        for file in files:
+            relative_path = os.path.join(root, file)
+            error_count += check_file(relative_path, file)
+    return error_count
+
+
+def check_file(relative_path: str, file: str) -> int:
+    """
+    Check the given file for any issues.
+
+    Writes error messages to the console, and returns the number of errors found
+
+    :param relative_path: The path of the file, including the directory, relative to the vault's root
+    :param file: The name of the file, without any directory
+    :return: The number of errors found in the given file 
+    """
+    errors = 0
+
+    if '.' not in file:
+        print(f'Error:\n  This file has no extension: consider adding ".md" to its name:\n  {relative_path} ')
```

Comment:
Perhaps it would be a good idea to decide on a universal "logging" method/syntax/manner. My own opinion is that 1 error = 1 line, in stead of 3.

Reply:
We sort of have a logging facility already, which is in `utils.py`.

However, this isn't logging, it's an ~~error~~ help  message that must never be suppressed.

> My own opinion is that 1 error = 1 line, in stead of 3.

Ah, do you mean the output should be on one line? It started like this, but was quite hard to parse, so I wrapped it up.

Yeah, I would have previously said the same, until I saw a tweet talking about the mindset of writing "help" messages rather than "error" messages... It completely changed how I phrased such things.

The output now looks like this:

![image](https://user-images.githubusercontent.com/4840096/148660444-423f918f-48ad-4a10-8cee-26207593fc37.png)

The path can get quite long - and so it's done with the file path on its own line, for glancability.
Similarly, the layout tries make it easy to spot the suggestion to add `.md`.

Reply:
Bear in mind that if this was on one line, it would wrap horribly in the GitHub Actions log.
And if there were several files missing their suffix, the single-line messages would be even harder to parse.